### PR TITLE
 	[8.0][MIG][website_crm_recaptcha] Backport from v9.

### PR DESCRIPTION
**Pull request by yajo**
Stuff related to `website_form` addon is removed, since it is unavailable in v8.

WIP until merged:
- [x] #231 

@Tecnativa.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Does this work also with the contact snippet?

File: website_crm_recaptcha/README.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+* Add site key to `recaptcha.key.site` system parameter
+* Add secret key to `recaptcha.key.secret` system parameter
+
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Go to the Contact form on Website
```
Comment (pedrobaeza) [inline website_crm_recaptcha/README.rst:25]: Add: `You will need to pass the Recaptcha challenge to submit the form.`

File: website_crm_recaptcha/README.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/186/8.0
+
+
+
+Known Issues / Road Map
```
Comment (pedrobaeza) [inline website_crm_recaptcha/README.rst]: Don't include an empty section

File: website_crm_recaptcha/i18n/en.po [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline website_crm_recaptcha/i18n/en.po]: Don't include this file

File: website_crm_recaptcha/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+from openerp.http import route
+from openerp.exceptions import ValidationError
+from openerp.addons.website_form_recaptcha.controllers.main import WebsiteForm
+from openerp.addons.website_crm.controllers.main import contactus
+
+
+class ContactUsRecaptcha(WebsiteForm, contactus):
+    @route()
+    def contactus(self, **kwargs):
```
Comment (pedrobaeza) [inline website_crm_recaptcha/controllers/main.py:13]: Why is this needed?

Comment (yajo) [inline website_crm_recaptcha/controllers/main.py:13]: In v9, [`website_crm` uses the controller from `website_form` to process the requests](https://github.com/odoo/odoo/blob/9.0/addons/website_crm/__openerp__.py#L12). V8 has its own, and does not have `website_form`, so this server-side validation must be added here.

File: website_crm_recaptcha/README.rst (hunk 1 above)
Comment (yajo) [inline website_crm_recaptcha/README.rst:25]: Done.

**Comment by yajo** [comment]
> Does this work also with the contact snippet?

What snippet do you mean?

**Comment by pedrobaeza** [comment]
This one: https://github.com/OCA/website/tree/8.0/website_snippet_contact_form

**Comment by lasley** [comment]
@pedrobaeza It would need to be implemented in the view, but it looks like the controller logic here would cover it in the snippet as well. V9 version would need about the same amount of work.

**Comment by yajo** [comment]
Theoretically it should work.

**Comment by rafaelbn** [comment]
Please @Yajo rebase https://github.com/OCA/website/pull/231 has been merged. Thanks

**Comment by lasley** [comment]
code + functional 👍 

Just in case it expires, valid tokens for Runbot:

Site
6Lc6IiYTAAAAAP_o_788-FI84b7zJMiax4SH_enF

Secret
6Lc6IiYTAAAAADOJX1PtcR-b7cm-STCY2LT2qBQl

**Comment by pedrobaeza** [comment]
:+1: 

Merging this one also.
